crates/rethnet_eth/src/remote/cacheable_method_invocation.rs

```diff
@@ -0,0 +1,154 @@
+use crate::remote::methods::{GetLogsInput, MethodInvocation, Percentiles, TransactionInput};
+use crate::remote::BlockSpec;
+use crate::U256;
+use revm_primitives::{Address, B256};
+
+/// These method invocations are hashable and can be potentially cached.
+#[derive(Clone, Debug, PartialEq, Eq, Hash)]
+// Using repr(u8) to let us assign a number to each variant which is important to make sure hashing
+// remains consistent if new variants are added or if variants are reordered.
+#[repr(u8)]
+pub(super) enum CacheableMethodInvocation<'a> {
+    /// eth_call
+    Call(&'a TransactionInput, &'a Option<BlockSpec>) = 0,
+    /// eth_chainId
+    ChainId() = 1,
+    /// eth_estimateGas
+    EstimateGas(&'a TransactionInput, &'a Option<BlockSpec>) = 2,
+    /// eth_feeHistory
+    FeeHistory(
```

Comment:
This wasn't possible for the RPC variants, as it needed to support a specific JSON format that required sequences, but here we can using names for the variant members